File: izinga-ordermanager/src/main/java/io/curiousoft/izinga/ordermanagement/service/UserProfileService.java
package io.curiousoft.izinga.ordermanagement.service;

import io.curiousoft.izinga.commons.model.ProfileRoles;
import io.curiousoft.izinga.commons.model.UserProfile;
import io.curiousoft.izinga.commons.repo.UserProfileRepo;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserProfileService extends ProfileServiceImpl<UserProfileRepo, UserProfile>{

    public UserProfileService(UserProfileRepo userProfileRepo) {
        super(userProfileRepo);
    }

    public UserProfile findUserByPhone(String phone) {
        return profileRepo.findByMobileNumber(phone);
    }

    public List<UserProfile> findByLocation(ProfileRoles role, double latitude, double longitude, double range) {
        double maxLong = longitude + range,
                minLong = longitude - range;
        double maxLat = latitude + range,
                minLat = latitude - range;

        return profileRepo.findByRoleAndLatitudeBetweenAndLongitudeBetween(role, minLat,
                maxLat, minLong, maxLong);
    }

    @Override
    public UserProfile create(UserProfile profile) throws Exception {
        if(profileRepo.existsByMobileNumber(profile.getMobileNumber()))
            throw new Exception("User with phone number " + profile.getMobileNumber() + " already exist.");
        return super.create(profile);
    }
}

File: izinga-usermanagement/src/main/java/io/curiousoft/izinga/usermanagement/users/UserProfileService.kt
package io.curiousoft.izinga.usermanagement.users

import io.curiousoft.izinga.commons.model.ProfileRoles
import io.curiousoft.izinga.commons.model.UserProfile
import io.curiousoft.izinga.commons.repo.UserProfileRepo
import org.springframework.stereotype.Service

@Service
class UserProfileService(userProfileRepo: UserProfileRepo) : ProfileServiceImpl<UserProfileRepo, UserProfile>(userProfileRepo) {
    
    fun findUserByPhone(phone: String): UserProfile? {
        return profileRepo.findByMobileNumber(phone)
    }

    fun findByLocation(role: ProfileRoles, latitude: Double, longitude: Double, range: Double): List<UserProfile>? {
        val maxLong = longitude + range
        val minLong = longitude - range
        val maxLat = latitude + range
        val minLat = latitude - range
        return profileRepo.findByRoleAndLatitudeBetweenAndLongitudeBetween(
            role, minLat,
            maxLat, minLong, maxLong
        )
    }

    @Throws(Exception::class)
    override fun create(profile: UserProfile): UserProfile {
        if (profileRepo.existsByMobileNumber(profile.mobileNumber!!)) throw Exception("User with phone number " + profile.mobileNumber + " already exist.")
        return super.create(profile)
    }
}